derive_client/utils/w3.py

```diff
@@ -169,33 +168,36 @@ def estimate_fees(w3, percentiles: list[int], blocks=20, default_tip=10_000):
 
 def iter_events(
     w3: Web3,
-    log_filter: LogFilter,
+    filter_params: dict,
+    *,
+    condition: Callable[[AttributeDict], bool] = lambda _: True,
     max_block_range: int = 10_000,
     poll_interval: float = 5.0,
     timeout: float | None = None,
 ) -> Generator[AttributeDict, None, None]:
     """Stream matching logs over a fixed or live block window. Optionally raises TimeoutError."""
 
-    filter_params = log_filter.filter_params.copy()  # return original in TimeoutError
+    filter_params = filter_params.copy()  # return original in TimeoutError
```

Comment:
TODO: fix this overwrite; we want a copy of the original for reporting in the `TimeoutError`